beacon_node/lighthouse_network/src/peer_manager/peerdb.rs

```diff
@@ -300,6 +300,7 @@ impl<E: EthSpec> PeerDB<E> {
             .filter(move |(_, info)| {
                 // We check both the metadata and gossipsub data as we only want to count long-lived subscribed peers
                 info.is_connected()
+                    && info.is_synced_or_advanced()
```

Comment:
Need to think again about this change - this is used to determine if we have enough good peers when starting discovery query:

https://github.com/sigp/lighthouse/blob/c8bbe479199ef09c5e226aa089f462cdfc9b60c5/beacon_node/lighthouse_network/src/service/mod.rs#L1093-L1103

So the condition to push a `PeerManagerEvent::DiscoverSubnetPeers` below must match this 

https://github.com/sigp/lighthouse/blob/a250daa29beb64349fdc326a7d0a76f02fcc592c/beacon_node/lighthouse_network/src/peer_manager/mod.rs#L969-L992

Otherwise it might result in pushing a `DiscoverSubnetPeers` and getting ignored by network.